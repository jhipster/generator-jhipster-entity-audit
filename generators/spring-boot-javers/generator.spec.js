import { beforeAll, describe, expect, it } from 'vitest';

import { defaultHelpers as helpers, result } from 'generator-jhipster/testing';

const SUB_GENERATOR = 'spring-boot-javers';
const SUB_GENERATOR_NAMESPACE = `jhipster-entity-audit:${SUB_GENERATOR}`;

describe('SubGenerator spring-boot-javers of entity-audit JHipster blueprint', () => {
  describe('run', () => {
    beforeAll(async function () {
      await helpers
        .run(SUB_GENERATOR_NAMESPACE)
        .withJHipsterConfig()
        .withOptions({
          ignoreNeedlesError: true,
          auditFramework: 'javers',
        })
        .withJHipsterLookup()
        .withParentBlueprintLookup();
    });

    it('should succeed', () => {
      expect(result.getStateSnapshot()).toMatchSnapshot();
    });
  });
  describe('should generate AuditedEntity', () => {
    beforeAll(async function () {
      await helpers
        .runJDL(
          `
application {
  config {
    baseName jhipster
    blueprints [generator-jhipster-entity-audit]
  }

  config(generator-jhipster-entity-audit) {
    auditFramework javers
  }

  entities *
}

@EnableAudit
@ChangelogDate("20200101000100")
entity CamelCase {
  name String required
}

@EnableAudit
@EntityPackage("custom")
@ChangelogDate("20200101000200")
entity WithEntityPackage {
  name String required
}
`,
        )
        .withParentBlueprintLookup();
    });

    it('generate .yo-rc.json content', () => {
      expect(result.getSnapshot('**/.yo-rc.json')).toMatchSnapshot();
    });
    it('generate AuditedEntity content', () => {
      expect(result.getSnapshot('**/AuditedEntity.java')).toMatchSnapshot();
    });
  });
});
