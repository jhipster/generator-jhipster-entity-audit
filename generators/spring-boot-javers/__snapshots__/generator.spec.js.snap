// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubGenerator spring-boot-javers of entity-audit JHipster blueprint > run > should succeed 1`] = `
{
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/GeneratedByJHipster.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/audit/JaversAuthorProvider.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/audit/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/domain/Authority.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/domain/audit/AuditedEntity.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/domain/audit/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/domain/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/JaversEntityAuditResource.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/dto/EntityAuditAction.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/dto/EntityAuditEvent.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/dto/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/package-info.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/domain/AssertUtils.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/domain/AuthorityAsserts.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/domain/AuthorityTest.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/domain/AuthorityTestSamples.java": {
    "stateCleared": "modified",
  },
}
`;

exports[`SubGenerator spring-boot-javers of entity-audit JHipster blueprint > should generate AuditedEntity > generate .yo-rc.json content 1`] = `
{
  ".yo-rc.json": {
    "contents": "{
  "generator-jhipster": {
    "baseName": "jhipster",
    "blueprints": [
      {
        "name": "generator-jhipster-entity-audit"
      }
    ],
    "entities": [
      "CamelCase",
      "WithEntityPackage"
    ],
    "jwtSecretKey": "SECRET--64",
    "languages": [
      "en"
    ],
    "nativeLanguage": "en"
  },
  "generator-jhipster-entity-audit": {
    "auditFramework": "javers",
    "auditPage": true
  }
}
",
    "stateCleared": "modified",
  },
}
`;

exports[`SubGenerator spring-boot-javers of entity-audit JHipster blueprint > should generate AuditedEntity > generate AuditedEntity content 1`] = `
{
  "src/main/java/com/mycompany/myapp/domain/audit/AuditedEntity.java": {
    "contents": "package com.mycompany.myapp.domain.audit;

public enum AuditedEntity {
    CAMELCASE(com.mycompany.myapp.domain.CamelCase.class, "domain.CamelCase"),
    CUSTOM_WITHENTITYPACKAGE(com.mycompany.myapp.custom.domain.WithEntityPackage.class, "custom.domain.WithEntityPackage");
    // jhipster-needle-add-audited-entities - JHipster will add entities to this enum

    private final Class<?> entityClass;
    private final String eventEntityType;

    private AuditedEntity(Class<?> entityClass, String eventEntityType) {
        this.entityClass = entityClass;
        this.eventEntityType = eventEntityType;
    }

    public Class<?> getEntityClass() {
        return entityClass;
    }

    public String getEventEntityType() {
        return eventEntityType;
    }
}
",
    "stateCleared": "modified",
  },
}
`;
